orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/DeploymentDetailsAware.groovy

```diff
@@ -52,9 +52,12 @@ trait DeploymentDetailsAware {
   Stage getPreviousStageWithImage(Stage stage, String targetRegion, String targetCloudProvider) {
     getAncestors(stage).find {
       def regions = (it.context.region ? [it.context.region] : it.context.regions) as Set<String>
-      (it.context.containsKey("ami") || it.context.containsKey("amiDetails")) &&
-      (!targetRegion || regions?.contains(targetRegion)) &&
-      (targetCloudProvider == it.context.cloudProvider || targetCloudProvider == it.context.cloudProviderType)
+      def cloudProviderFromContext = it.context.cloudProvider ?: it.context.cloudProviderType
+      boolean hasTargetCloudProvider = !cloudProviderFromContext || targetCloudProvider == cloudProviderFromContext
+      boolean hasTargetRegion = !targetRegion || regions?.contains(targetRegion)
+      boolean hasImage = it.context.containsKey("ami") || it.context.containsKey("amiDetails")
```

Comment:
I don't think so. The sample pipeline we tested with used ami, and so did
the earlier samples I had.

On Aug 8, 2016 4:25 PM, "Tomás Lin" <notifications@github.com> wrote:

> In orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/kato/tasks/
> DeploymentDetailsAware.groovy
> <https://github.com/spinnaker/orca/pull/923#discussion_r73947267>:
>
> > @@ -52,9 +52,12 @@ trait DeploymentDetailsAware {
> >    Stage getPreviousStageWithImage(Stage stage, String targetRegion, String targetCloudProvider) {
> >      getAncestors(stage).find {
> >        def regions = (it.context.region ? [it.context.region] : it.context.regions) as Set<String>
> > -      (it.context.containsKey("ami") || it.context.containsKey("amiDetails")) &&
> > -      (!targetRegion || regions?.contains(targetRegion)) &&
> > -      (targetCloudProvider == it.context.cloudProvider || targetCloudProvider == it.context.cloudProviderType)
> > +      def cloudProviderFromContext = it.context.cloudProvider ?: it.context.cloudProviderType
> > +      boolean hasTargetCloudProvider = !cloudProviderFromContext || targetCloudProvider == cloudProviderFromContext
> > +      boolean hasTargetRegion = !targetRegion || regions?.contains(targetRegion)
> > +      boolean hasImage = it.context.containsKey("ami") || it.context.containsKey("amiDetails")
>
> should this be amiName?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/spinnaker/orca/pull/923/files/d8dc45cdb299ce435ec022f8a8dbb6d289cbac38#r73947267>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIC_1obzyWBpHOPHuTL3-9rT5SjqoDAdks5qd5DKgaJpZM4Jfa-D>
> .
>